src/meta/meta_ckpt.c

```diff
@@ -209,8 +210,32 @@ __ckpt_set(WT_SESSION_IMPL *session, const char *fname, const char *v, bool use_
     if (use_base && session->dhandle != NULL) {
         WT_ERR(__wt_scr_alloc(session, 0, &tmp));
         WT_ASSERT(session, strcmp(session->dhandle->name, fname) == 0);
+
+        /* Check the metadata is not corrupted. */
+        meta_base = session->dhandle->meta_base;
+        meta_base_length = meta_base == NULL ? 0 : strlen(meta_base);
+#ifdef HAVE_DIAGNOSTIC
+        if (session->dhandle->orig_meta_base_length != meta_base_length ||
+          (session->dhandle->orig_meta_base == NULL && meta_base != NULL) ||
```

Comment:
I think we can simplify it (see below about moving the length check and comment above that I think meta_base is not NULL. It can just be `if (!WT_STREQ(...))`